client/controller/controller.js

```diff
@@ -0,0 +1,355 @@
+
+app.controller('AppCtrl',function($scope, $http, filterdata) {
+
+
+    // ----> Code from autocomplete.js <------
+    var inputFrom = document.getElementById('input');
+
+
+    var autocompleteFrom = new google.maps.places.Autocomplete(inputFrom);
+    google.maps.event.addListener(autocompleteFrom, 'place_changed', function() {
+
+
+        var place = autocompleteFrom.getPlace();
+
+        var placeName = place.formatted_address;
+
+        $scope.place = placeName;
+
+
+
+    }); //end of autocomplete function.
+
+
+    //Funtion for the search button
+    // This function will recieve data from the server, filter the data , and respond back to the DOM
+    $scope.search = function() {
+
+     // newTyped();
+
+      //call the geocode function
+      $scope.geocode($scope.place);
+
+
+    };
+
+
+    $scope.geocode = function(placeName){
+
+
+      var geocoder = new google.maps.Geocoder();
+      geocoder.geocode({
+              address: placeName
+            }, function(results, status){
+              if(status === google.maps.GeocoderStatus.OK){
+
+                address = results[0];
```

Comment:
result.length is always > 0. I am using the first element in the array because that is the one we need. Google searches all kinds of places releted to the query we requested.